Partial UT run is reported as an error

| | |
|:---|:---|
|**_F´ Version_**| |
|**_Affected Component_**|   |
---
## Problem Description

Using the following command:
```
cd fprime/Drv
fprime-util check
```

UTs in Drv pass, but the framework reports an error because not all UTs registered by the entire framwork have been ran

```
[...]
92/93 Test #92: Utils_ut_exe .....................................***Not Run   0.00 sec
      Start 93: Types_Circular_Buffer_ut_exe
Could not find executable /home/chammard/LENG-Work/fprime/fprime/build-fprime-automatic-native-ut/bin/Linux/Types_Circular_Buffer_ut_exe
93/93 Test #93: Types_Circular_Buffer_ut_exe .....................***Not Run   0.00 sec

2% tests passed, 91 tests failed out of 93


The following tests FAILED:
	  1 - Fw_Buffer_ut_exe (Not Run)
	  2 - Fw_Dp_ut_exe (Not Run)
[....]

	 92 - Utils_ut_exe (Not Run)
	 93 - Types_Circular_Buffer_ut_exe (Not Run)
Errors while running CTest
Output from these tests are in: /home/chammard/LENG-Work/fprime/fprime/build-fprime-automatic-native-ut/Testing/Temporary/LastTest.log
Use "--rerun-failed --output-on-failure" to re-run the failed cases verbosely.
[ERROR] Command '['ctest', '--test-dir', '/home/chammard/LENG-Work/fprime/fprime/build-fprime-automatic-native-ut', '--no-tests=error']' returned non-zero exit status 8.
```

It appears to be trying to run all UTs registered in the framework.

## Context / Environment


```
Operating System: Linux
CPU Architecture: x86_64
Platform: Linux-4.18.0-553.64.1.el8_10.x86_64-x86_64-with-glibc2.28
Python version: 3.9.20
CMake version: 3.26.0
Pip version: 25.0.1
Pip packages:
    fprime-tools==4.0.1
    fprime-gds==4.0.1
    fprime-fpp==3.0.0
Project submodules:
```

## Expected Behavior

CTest should either:
- only attempt to run the UTs that are registered within `Drv` if `--recursive` is given
- not run any UTs

--- a/src/fprime/fbuild/check.py
+++ b/src/fprime/fbuild/check.py
@@ -70,6 +70,16 @@ class Check(EnumeratedAction):
             context and context != ["all"] and ".*" not in context
         ):
             cli_args.append("-V")
+
+        if not context:
+            # This only happens if the provided path does not contain tests
+            # and neither '--all' nor '--recursive' were provided
+            print(
+                "[INFO] No tests were found in the given context.",
+                "Did you mean to use `fprime-util check --recursive` or `--all` ?",
+            )
+            return
+
         # When not "all" append a regex to filter tests. .* works as a regex
         if context and context != ["all"]:
             test_regex = f"^({'|'.join(context)})$"

--- a/src/fprime/fbuild/check.py
+++ b/src/fprime/fbuild/check.py
@@ -9,7 +9,7 @@ import subprocess
 import sys
 from typing import Tuple, Dict, List
 
-from fprime.fbuild.types import NoSuchTargetException
+from fprime.fbuild.types import NoTargetFoundException
 from fprime.fbuild.target import (
     TargetContext,
     TargetScope,
@@ -55,7 +55,7 @@ class Check(EnumeratedAction):
         if not context:
             # This only happens if the provided path does not contain tests
             # and neither '--all' nor '--recursive' were provided
-            raise NoSuchTargetException(
+            raise NoTargetFoundException(
                 "No tests were found in the given context. Did you mean to "
                 "use `fprime-util check --recursive` or `--all` ?"
             )

--- a/src/fprime/fbuild/check.py
+++ b/src/fprime/fbuild/check.py
@@ -50,6 +50,16 @@ class Check(EnumeratedAction):
         args: Tuple[Dict[str, str], List[str], Dict[str, bool]],
     ):
         """Execute this target"""
+
+        if not context:
+            # This only happens if the provided path does not contain tests
+            # and neither '--all' nor '--recursive' were provided
+            print(
+                "[ERROR] No tests were found in the given context.",
+                "Did you mean to use `fprime-util check --recursive` or `--all` ?",
+            )
+            sys.exit(1)
+
         cli_args = [
             self.EXECUTABLE,
             "--test-dir",
@@ -70,15 +80,6 @@ class Check(EnumeratedAction):
             context and context != ["all"] and ".*" not in context
         ):
             cli_args.append("-V")
-
-        if not context:
-            # This only happens if the provided path does not contain tests
-            # and neither '--all' nor '--recursive' were provided
-            print(
-                "[INFO] No tests were found in the given context.",
-                "Did you mean to use `fprime-util check --recursive` or `--all` ?",
-            )
-            return
 
         # When not "all" append a regex to filter tests. .* works as a regex
         if context and context != ["all"]:

--- a/src/fprime/fbuild/check.py
+++ b/src/fprime/fbuild/check.py
@@ -9,6 +9,7 @@ import subprocess
 import sys
 from typing import Tuple, Dict, List
 
+from fprime.fbuild.types import NoSuchTargetException
 from fprime.fbuild.target import (
     TargetContext,
     TargetScope,
@@ -54,11 +55,10 @@ class Check(EnumeratedAction):
         if not context:
             # This only happens if the provided path does not contain tests
             # and neither '--all' nor '--recursive' were provided
-            print(
-                "[ERROR] No tests were found in the given context.",
-                "Did you mean to use `fprime-util check --recursive` or `--all` ?",
+            raise NoSuchTargetException(
+                "No tests were found in the given context. Did you mean to "
+                "use `fprime-util check --recursive` or `--all` ?"
             )
-            sys.exit(1)
 
         cli_args = [
             self.EXECUTABLE,

--- a/src/fprime/fbuild/types.py
+++ b/src/fprime/fbuild/types.py
@@ -36,6 +36,10 @@ class MissingBuildCachePath(FprimeException):
     """An exception indicating that a path in the build cache is missing"""
 
 
+class NoTargetFoundException(FprimeException):
+    """An exception indicating that no target was found in the given context"""
+
+
 class BuildType(Enum):
     """
     An enumeration used to represent the various build types used to build fprime. These types can support different

